src/ros2cs/ros2cs_core/CMakeLists.txt

```diff
@@ -29,6 +29,8 @@ find_package(ament_cmake_export_assemblies REQUIRED)
 find_package(ament_cmake REQUIRED)
 find_package(dotnet_cmake_module REQUIRED)
 
+set(CSHARP_TARGET_FRAMEWORK "netstandard2.1")
```

Comment:
I choose `netstandard2.1` since that was written in [ros2cs_core.csproj](https://github.com/RobotecAI/ros2cs/blob/master/src/ros2cs/ros2cs_core/ros2cs_core.csproj) and without specifying it explicitly there would be errors because of a DLL version mismatch, maybe because Unity shipped different DLLs than the ones required by the default standard choosen by the build system.
I wil check your suggestion.

Reply:
> I choose netstandard2.1 since that was written in [ros2cs_core.csproj](https://github.com/RobotecAI/ros2cs/blob/master/src/ros2cs/ros2cs_core/ros2cs_core.csproj) and without

I see, I need to double-check that csproj in the near future.

> I tried using netcoreapp but the build failed because of something requiring netcoreapp1.0.

Yes, the "netcoreapp" is "core" and misses some implementations. 

I think going with `netstandard2.0` and proposed changes is a way to go, lets's do that :+1: